pexec/managed_process.go

```diff
@@ -35,6 +35,11 @@ type ManagedProcess interface {
 	// Stop signals and waits for the process to stop. An error is returned if
 	// there's any system level issue stopping the process.
 	Stop() error
+
+	// Returns the oldest buffered log line. Up to 1000 log lines are buffered, with the
```

Comment:
Should I make the buffer size configurable?

Reply:
what about having ProcessConfig have a io.Writer for StdOut + StdErr that you manage? that gives you more options. This current method feels a little forced into to support the SLAM use case.

Reply:
I thought everything in ProcessConfig has to be serializable as json.

Reply:
I could add more arguments to NewManagedProcess() to pass in a io.Writer, but that seemed invasive.

Reply:
nope doesn't have to serialize necessarily. can mark those fields as `json:"-"`